astrogrism/simulate/simulate.py

```diff
@@ -96,3 +102,116 @@ def generate_synthetic_spectrum(grism, detector=None, temp_path=gettempdir(), ve
+        TBF: Should be replaced with a Grism identifier only in the future
+
+    wide_field_image : numpy.ndarray or astropy.io.fits.ImageHDU
+        The image or array to disperse the given spectrum atop
+
+    spectrum : specutils.Spectrum1D
+        The spectrum to diperse the on the supplied image
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+    """
+    grismobs = GrismObs(grism_file)
+    if detector:
+        image2grism = grismobs.geometric_transforms[f'CCD{detector}'].get_transform('detector',
+                                                                                    'grism_detector'
+                                                                                    )
+    else:
+        image2grism = grismobs.geometric_transforms.get_transform('detector', 'grism_detector')
+
+    if type(wide_field_image) is ImageHDU:
+        data = wide_field_image.data
+    else:
+        data = wide_field_image
+
+    shape = data.shape
+    simulated_data = np.zeros(shape)
+
+    # Start progressbar
+    bar = progressbar.ProgressBar(maxval=shape[0],
+                                  widgets=[progressbar.Bar('=', '[', ']'),
+                                           ' ',
+                                           progressbar.Percentage()
+                                           ]
+                                  )
+    bar.start()
+
+    # For each pixel in the science image, we need to disperse it's spectrum
+    for x_pixel in range(0, shape[0]):
+        bar.update(x_pixel)
+        for y_pixel in range(0, shape[1]):
+            # Get the flux of the science pixel
+            # We'll need to scale the spectrum to this brightness later
+            data_flux = data[x_pixel][y_pixel]
+            if np.isnan(data_flux) or (data_flux == 0):
+                continue
+
+            # For each Wavelength in the spectrum,
+            # calculate where, in pixels, that wavelength would fall on the detector
+            dispersed_coords = image2grism(x_pixel, y_pixel, spectrum.wavelength, 1)
+            for step in range(0, len(spectrum.wavelength)):
+                spectrum_flux = spectrum.flux[step]
+
+                dispersed_x = dispersed_coords[0][step]
+                dispersed_y = dispersed_coords[1][step]
+
+                # If the dispersed position of the wavelength is inside the bounds of the image,
+                # write the spectrum
+                if (0 < dispersed_x < simulated_data.shape[0]) and (0 < dispersed_y < simulated_data.shape[1]): # noqa
+                    # Scale the flux of the spectrum to the brightness of the original pixel
+                    # TBF: Substitute floor with proper subpixel drizzling
+                    simulated_data[floor(dispersed_x)][floor(dispersed_y)] += (data_flux * spectrum_flux).value # noqa
```

Comment:
Should the simulated data come out with the same units as the input direct image (counts, flux, whatever)? Here it looks l like the output will be unitless.